internal/scheduler/pool_assigner.go

```diff
@@ -0,0 +1,169 @@
+package scheduler
+
+import (
+	"context"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/armadaproject/armada/internal/armada/configuration"
+	protoutil "github.com/armadaproject/armada/internal/common/proto"
+	"github.com/armadaproject/armada/internal/scheduler/database"
+	"github.com/armadaproject/armada/internal/scheduler/jobdb"
+	"github.com/armadaproject/armada/internal/scheduler/schedulerobjects"
+)
+
+// PoolAssigner allows jobs to be assigned to a pool
+// Note that this is intended only for use with metrics calculation
+type PoolAssigner interface {
```

Comment:
lol- yes it would, except the way we define pools right now makes this impossible

Reply:
To be clear- the way Armada represents pools right now is completely dynamic.  Inside GR we restrict this in config so that it's basically (although not entirely) static, but that doesn't apply in the general case.  

I don't think the way we do pools right now is great- and stuff like this (having to reassign the pool on every metrics run) is the result, but in order to fix this, I think we'd have to redefine how Pools work  and that's a much bigger task.